### DISCUSSION TITLE
Advice about where to start: Remembering previous conversations critical for this use case

### DISCUSSION THREAD

**Discussion by philiprhoades** [facebookresearch/ParlAI#4680] [General]
People,

I need to develop (eventually AI) Avatars for myself, my (deceased and brain-frozen) parents, other relatives, friends and members of my non-profit Neural Archives Foundation:

http://neuralarchivesfoundation.org

For this use case increasingly sophisticated LMMs are nice but my first priority is having an entity that remembers ALL its previous conversations with ALL the different people it talks to is mission critical . . maybe this is not possible now but soon?

The second priority for the Avatars is bulk uploading of personal information from written textual information ie it would need semantic parsing and turning into whatever storage structure is used.

So, advice about where to start would be enormously appreciated! - my background is in Fortran IV (wayback) then C, UNIX, Linux and more Ruby more recently.

Thanks,

Phil.

**Comment by mojtaba-komeili** [comment]
First of all, I should tell you not to have very high expectations from the current state of conversation chatbots. They may act OK half the time and be completely off the other half. Specially forcing them to focus on such narrow domain is very tricky.
With that, probably having a dataset of their conversations, and fine-tuning a conversation model such as blenderbot2 might be a reasonable starting point.

**Reply by philiprhoades** [reply]
> First of all, I should tell you not to have very high expectations from the current state of conversation chatbots. They may act OK half the time and be completely off the other half.

? Really ? - I thought they were getting so good that there were recent claims that one of them had become sentient / conscious . .

> Specially forcing them to focus on such narrow domain is very tricky.

I can see that would be more of an issue - but I am thinking that that part would be in ADDITION to a general large model dataset . . 

> With that, probably having a dataset of their conversations,

Not sure what you mean by that - for my specific needs - I want to somehow semantically parse all my own documents, correspondence, emails, social media stuff etc as well as add the individual project Q&As.

> and fine-tuning a conversation model such as blenderbot2 might be a reasonable starting point.

Well BB2 with its memory is the obvious target to get started with it seems to me . .

**Reply by mojtaba-komeili** [reply]
Unfortunately there is a lot of false information and claims about the performance of chatbot, which are not close to true.
About parsing and the exact process, it sounds like a real project that is beyond what can be discussed in this simple comment thread, and the main focus of our research team.
